Improve content selection (#70)
ambanum/CGUs: CONTRIBUTING.md

  "name": "<the public name of the service>",
  "documents": {
    "<document type>": {

>       "fetch": "<an array or service specific filter function name>",

Reviewer: ```suggestion
      "fetch": "<the URL where the document can be found>",
```

>       "filter": "<an array of service specific filter function name>",

Reviewer: ```suggestion
      "filter": "<an array of service specific filter function names>",
```

>       "filter": "<a selector or an array of selector that targets the meaningful parts of the document, excluding elements such as headers, footers and navigation>",

Reviewer: ```suggestion
      "filter": "<a CSS selector, a range selector or an array of selectors that target the meaningful parts of the document, excluding elements such as headers, footers and navigation>",
```

>       "remove": "<a selector or an array of selector that targets the noise parts of the document that has to be removed. Useful to remove parts that are inside the selected parts>",

Reviewer: ```suggestion
      "remove": "<a CSS selector, a range selector or an array of selectors that target the noise parts of the document that have to be removed. Useful to remove parts that are inside the selected parts>",
```

>       "select": "<a selector or an array of selector that targets the meaningful parts of the document, excluding elements such as headers, footers and navigation>",

Reviewer: ```suggestion
      "select": "<a CSS selector, a range selector or an array of selectors that target the meaningful parts of the document, excluding elements such as headers, footers and navigation>",
```

Each filter is exposed as a named function export that takes a `document` parameter and behaves like the `document` object in a browser DOM.

> It is actually a [JSDOM](https://github.com/jsdom/jsdom) document instance.

> Learn more about [usual noises](https://github.com/ambanum/CGUs/wiki/Usual-noise).

Reviewer: ```suggestion
Learn more about [usual noise](https://github.com/ambanum/CGUs/wiki/Usual-noise).
```

### Selectors

> A selector allows to target a part of the document. It can be of two types, a **css selector** or a **range selector**.

Reviewer: > Add a link to define CSS selector?

> A range selector is defined with a _start_ and an _end_ css selector. It is also necessary to define if the range starts before or after the element targeted by the _start_ css selector and to defined if it ends

Reviewer: ```suggestion
A range selector is defined with a _start_ and an _end_ css selector. It is also necessary to define if the range starts before or after the element targeted by the _start_ css selector and to define if it ends
```

> before or after the element targeted by the _end_ css selector. To do that, four keys are available to properly describe a range selector `startBefore`, `startAfter`, `endBefore` and `endAfter`.

Reviewer: ```suggestion
before or after the element targeted by the _end_ css selector. To do that, four keys are available to properly describe a range selector: `startBefore`, `startAfter`, `endBefore` and `endAfter`.
```

> Thus, a range selector look like this:

Reviewer: ```suggestion
Thus, a range selector looks like this:
```

```json
{
  "start[Before|After]": "<css selector>",
  "end[Before|After]": "<css selector>"
}
```

For example, the following selector will select the content between the element targeted by the css selector `#privacy-eea`, including it, and the element targeted by the css selector `#privacy-row`, excluding it:

```json
{
  "startBefore": "#privacy-eea",
  "endBefore": "#privacy-row"
}
```

### Selecting jurisdiction and language

For now, when multiple versions coexist, terms are only tracked in their English version and for the European (EEA) jurisdiction.

## Test your declaration

Test the declaration by running the following command with the service id:

```
npm run validate $service_id
```

> The service id is the case sensitive name of the service declaration file without the extension. For example, for `Twitter.json`, the service id is `Twitter`.

> You can also only validate the schema structure:

Reviewer: ```suggestion
Since this operation fetches documents and could be long, you can also validate the declaration structure only:
```
